vllm/v1/outputs.py

```diff
@@ -114,6 +115,15 @@ class ModelRunnerOutput:
     num_nans_in_logits: Optional[dict[str, int]] = None
 
 
+# ModelRunnerOutput wrapper for async scheduling. Contains device tensors
+# which must be copied to the host before sending to the scheduler process.
+class AsyncModelRunnerOutput(ABC):
+
+    @abstractmethod
+    def copy_to_host(self) -> ModelRunnerOutput:
+        pass
```

Comment:
Would be good to add docstring here too, including warning that this should only be called once.

I wonder if calling it something like `get_output()` or `materialize()` would be better, since in abstract context the fact that this is doing a copy isn't so relevant.